File: backend/app/schemas/company.py
from datetime import datetime
from pydantic import BaseModel, HttpUrl, ConfigDict


class CompanyBase(BaseModel):
    name: str
    website: HttpUrl | None = None
    industry: str | None = None
    country: str | None = None


class CompanyCreate(CompanyBase):
    pass


class CompanyUpdate(CompanyBase):
    name: str | None = None
    website: HttpUrl | None = None
    industry: str | None = None
    country: str | None = None


class Company(CompanyBase):
    id: int
    last_updated_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/company.ts
export type Company = {
  id: number
  name: string
  website?: string | null
  industry?: string | null
  country?: string | null
  last_updated_at?: string | null
  created_at: string
  updated_at: string
}

export type CompanyCreate = {
  name: string
  website?: string
  industry?: string
  country?: string
}
